src/components/ConnectNode/ConnectNode.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { FC, useState, memo, useCallback, ChangeEvent } from 'react';
+import { SidebarButton } from 'components/Sidebar';
+import DropDown from 'components/elements/DropDown';
+import { Pathway, PathwayNode } from 'pathways-model';
+import useStyles from './styles';
+import { addTransition } from 'utils/builder';
+import { usePathwaysContext } from 'components/PathwaysProvider';
+
+import { faLevelDownAlt } from '@fortawesome/free-solid-svg-icons';
+import { Button } from '@material-ui/core';
+import Tooltip from '@material-ui/core/Tooltip';
+
+interface ConnectNodeButtonProps {
+  pathway: Pathway;
+  rootNode: PathwayNode;
+}
+
+const getAncestorNodes = (
+  pathway: Pathway,
+  rootNodeKey: string,
+  currNodeKey: string
+): Array<PathwayNode> => {
+  let ancestors: Array<PathwayNode> = [];
+  let previousNodes: Array<PathwayNode> = [];
+  const currNode = pathway.nodes[currNodeKey];
+
+  for (const transition of currNode.transitions) {
+    if (transition.transition === rootNodeKey) ancestors.push(currNode);
+    else {
+      previousNodes = getAncestorNodes(
+        pathway,
+        rootNodeKey,
+        pathway.nodes[transition.transition].key ?? ''
+      );
+      if (previousNodes !== undefined && previousNodes.length > 0) {
+        if (!ancestors.some(node => node === currNode)) ancestors.push(currNode);
+        ancestors = ancestors.concat(previousNodes);
+      }
+    }
+  }
```

Comment:
This can clean up to be
```suggestion
 currNode.transitions.forEach((transition) => {
    if (transition.transition === rootNodeKey) ancestors.push(currNode);
    else {
      previousNodes = getAncestorNodes(pathway, rootNodeKey, transition.transition);
      if (previousNodes?.length) {
        if (!ancestors.some(node => node.key === currNode.key)) ancestors.push(currNode);
        ancestors = ancestors.concat(previousNodes);
      }
    }
  }
```